### PR TITLE
Fix graphite webapp log directory permissions

### PR DESCRIPTION
**Pull request by kiiranh**
This PR fixes Issue #316.

### CODE REVIEW

**Comment by cbaenziger** [comment]
Updated, found a related issue recorded at Chef-BACH #316.

File: cookbooks/bcpc/recipes/graphite.rb [hunk 1]
```diff
@@ -113,6 +113,17 @@
   recursive true
 end
 
+directory "#{node['bcpc']['graphite']['local_log_dir']}/webapp" do
+  owner "www-data"
+  group "www-data"
+end
+
+["info.log", "exception.log" ].each do |f|
+  file "#{node['bcpc']['graphite']['local_log_dir']}/webapp/#{f}" do
```
Comment (pu239ppy) [inline cookbooks/bcpc/recipes/graphite.rb]: I don't think you need to "touch" the files

Comment (kiiranh) [inline cookbooks/bcpc/recipes/graphite.rb]: On a new cluster, the "webapp" directory and files "info.log" and "exception.log" are owned by root. Unless something touches the files, webapp will never be able to write to those files.

**Comment by amithkanand** [comment]
For some reason, I do not see any permissions with directory/file structure. I am also able to see data on `graphite` UI.

**Comment by cbaenziger** [comment]
Looks plausible to me. Please report on how testing goes.

**Comment by kiiranh** [comment]
Confirmed that the problem persists on a fresh chef-bach master build without any modifications:

```bash
ubuntu@bcpc-vm1:/opt/graphite$ ls -lrt
total 28
drwxrwxr-x 4 root     root     4096 Nov  3 15:05 lib
drwxrwxr-x 4 root     root     4096 Nov  3 15:05 webapp
drwxrwxr-x 3 root     root     4096 Nov  3 15:05 examples
drwxrwxr-x 2 root     root     4096 Nov  3 15:05 bin
drwxr-xr-x 7 root     root     4096 Nov  3 15:05 static
drwxrwxr-x 2 root     root     4096 Nov  3 15:48 conf
drwxrwxr-x 7 www-data www-data 4096 Nov  3 15:49 storage
ubuntu@bcpc-vm1:/opt/graphite$ cd storage/
ubuntu@bcpc-vm1:/opt/graphite/storage$ ls -lrt
total 28
drwxrwxr-x 2 root     root     4096 Nov  3 14:22 rrd
drwxrwxr-x 2 root     root     4096 Nov  3 14:22 lists
drwxrwxr-x 2 root     root     4096 Nov  3 14:22 ceres
drwxrwxr-x 5 www-data www-data 4096 Nov  3 15:11 log
drwxrwxr-x 6 www-data www-data 4096 Nov  3 15:19 whisper
-rw-r--r-- 1 root     root        5 Nov  3 15:49 carbon-cache-a.pid
-rw-r--r-- 1 root     root        5 Nov  3 15:49 carbon-relay-a.pid
ubuntu@bcpc-vm1:/opt/graphite/storage$ cd log/
ubuntu@bcpc-vm1:/opt/graphite/storage/log$ ls -lrt
total 12
drwxrwxr-x 2 root root 4096 Nov  3 15:05 webapp
drwxr-xr-x 3 root root 4096 Nov  3 15:11 carbon-relay
drwxr-xr-x 2 root root 4096 Nov  3 15:11 carbon-cache-a
ubuntu@bcpc-vm1:/opt/graphite/storage/log$ ls -lrt webapp/
total 0
-rw-r--r-- 1 root root 0 Nov  3 15:05 info.log
-rw-r--r-- 1 root root 0 Nov  3 15:05 exception.log
ubuntu@bcpc-vm1:/opt/graphite/storage/log$
```

**Pull request merged**